src/preferences.cpp

```diff
@@ -152,24 +156,68 @@ Preferences::Preferences(QWidget *parent) :
 	});
 
 	ui->label_restart->setVisible(false);
-	ui->languageCombo->addItems(getLanguageList());
+	ui->languageCombo->addItems(getOptionsList());
 
 	QString preference_ini_file = getPreferenceIniFile();
 	QSettings settings(preference_ini_file, QSettings::IniFormat);
 
 	pref_api->setObjectName(QString("Preferences"));
 	pref_api->load(settings);
-
 	connect(ui->languageCombo, &QComboBox::currentTextChanged, [=](QString lang) {
-		language = lang;
-		if (m_initialized)
-			ui->label_restart->setVisible(true);
-		else
-			m_initialized = true;
-		Q_EMIT notify();
+			if(lang=="browse"){
+				QString langtemp = loadLanguage();
+				if(!langtemp.isEmpty()){
+					language=langtemp;
+					ui->languageCombo->addItem(language);
+					ui->languageCombo->setCurrentText(language);
+					if (m_initialized)
+						ui->label_restart->setVisible(true);
+					else
+						m_initialized = true;
+						}
+				else{
+					if(!getLanguageList().contains(language)){
+						QFileInfo info(language);
+						language = info.fileName().remove(".qm");
+						}
+					ui->languageCombo->setCurrentText(language);
+					ui->label_restart->setVisible(false);
+				}
+				}
+			else{
+				language = lang;
+				if (m_initialized)
+					ui->label_restart->setVisible(true);
+				else
+					m_initialized = true;
+        			}
+			Q_EMIT notify();
 	});
 }
 
+QStringList Preferences::getOptionsList()
+{   
+	QFile f(getPreferenceIniFile());
+	f.open(QFile::ReadOnly);
+    QTextStream in (&f);
+	const QStringList content = in.readAll().split("\n");
+	QString language;
+	for(auto s : content) {
+		if(s.startsWith("language")){
+			language=s.split("=")[1].trimmed();
```

Comment:
I think there already is a property called language for this that uses `getLanguage`, `setLanguage`, is the parsing of the ini file needed?